java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/SinkStreamObserver.java

```diff
@@ -151,22 +138,29 @@ public void onNext(ConnectorServiceProto.SinkStreamRequest sinkTask) {
                             .withDescription("Epoch is not started. Invoke `StartEpoch` first.")
                             .asRuntimeException();
                 }
-                if (sinkTask.getSync().getEpoch() != currentEpoch) {
+                if (sinkTask.getBarrier().getEpoch() != currentEpoch) {
                     throw INVALID_ARGUMENT
                             .withDescription(
                                     "invalid epoch: expected sync to epoch "
                                             + currentEpoch
                                             + ", got "
-                                            + sinkTask.getSync().getEpoch())
+                                            + sinkTask.getBarrier().getEpoch())
                             .asRuntimeException();
                 }
-                sink.sync();
-                currentEpoch = sinkTask.getSync().getEpoch();
-                LOG.debug("Epoch {} synced", currentEpoch);
-                responseObserver.onNext(
-                        ConnectorServiceProto.SinkResponse.newBuilder()
-                                .setSync(SyncResponse.newBuilder().setEpoch(currentEpoch).build())
-                                .build());
+                boolean isCheckpoint = sinkTask.getBarrier().getIsCheckpoint();
+                sink.barrier(isCheckpoint);
+                currentEpoch = sinkTask.getBarrier().getEpoch();
+                LOG.debug("Epoch {} barrier {}", currentEpoch, isCheckpoint);
+                if (isCheckpoint) {
+                    responseObserver.onNext(
```

Comment:
Why only send back response for barrier message? What about other messages?

Reply:
On CN side, in current code we have to wait for the response for begin epoch and write batch request and we have to send all the request one by one, which is unnecessary. In this PR, we change to only respond for checkpoint barrier, and in this way, the remote sink writer on CN can process and send multiple stream chunk asynchronously without waiting for the connector node to finish processing each message.